### ISSUE TITLE
Move tests to pytest

### ISSUE THREAD

**Issue by thedrow** [azazel75/macropy#4]
pytest is a much better testing framework than unittest.
We need to migrate our tests to that framework so that we can test, benchmark etc.

**Comment by azazel75** [comment]
Yes, when we will have some more time. BTW, pytest should be able to drive unittest based tests, but I never tried it. See https://docs.pytest.org/en/latest/unittest.html
